code/game/objects/items/breakable.dm

```diff
@@ -0,0 +1,263 @@
+	var/glances_text	//String or list of strings when the item is attacked but the attack glances off. eg. "bounces" becomes "but it bounces off!"
+	var/breaks_text		//Visible message when the items breaks. eg. "breaks apart"
+	//Sounds:
+	var/breaks_sound	//Path to audible sound when the item breaks apart.
+	var/damaged_sound	//Path to audible sound when the item is damaged but not fully destroyed.
+	var/glanced_sound	//Path to audible sound when the item recives a glancing attack not strong enough to damage it.
+
+/obj/item/proc/on_broken(var/atom/target, var/range, var/speed, var/override, var/fly_speed, var/atom/hit_atom) //Called right before an object breaks.
+	//Drop and and propel any fragments:
+	drop_fragments(target,range,speed,override,fly_speed)
+	//Drop and propel any contents:
+	drop_contents(target,range,speed,override,fly_speed)
+	//Spill any reagents:
+	spill_reagents(hit_atom)
+	if(breaks_text)
+		visible_message("<span class='warning'>\The [src] [breaks_text]!</span>")
+	if(breaks_sound)
+		playsound(src, breaks_sound, 50, 1)
+	else if(damaged_sound)
+		playsound(src, damaged_sound, 50, 1)
+
+/obj/item/proc/drop_fragments(var/atom/target, var/range, var/speed, var/override, var/fly_speed) //Separate proc in case special stuff happens with a given item's fragments. Parameters are for throwing the fragments.
+	if(breakable_fragments.len)
+		var/oneeach=(isnull(fragment_amounts) || breakable_fragments.len != fragment_amounts.len) //default to 1 of each fragment type if fragement_amounts isn't specified or there's a length mismatch
+		var/numtodrop
+		var/thisfragment
+		for(var/frag_ind in 1 to breakable_fragments.len)
+			if(oneeach)
+				numtodrop=1
+			else
+				numtodrop=fragment_amounts[frag_ind]
+			thisfragment=breakable_fragments[frag_ind]
+			for(var/n in 1 to numtodrop)
+				var/obj/item/O = new thisfragment (get_turf(src))
+				//Transfer fingerprints, fibers, and bloodstains to the fragment.
+				transfer_fingerprints(src,O)
+				transfer_item_blood_data(src,O)
+				if(target && range && speed) //Propel the fragment if specified.
+					O.throw_at(target,range,speed,override,fly_speed)
+
+/obj/item/proc/drop_contents(var/atom/target, var/range, var/speed, var/override, var/fly_speed) //Drop the contents of the item and propel them if the item itself received a propulsive blow.
+	if(contents.len)
+		for(var/obj/item/thiscontent in contents)
+			thiscontent.loc = src.loc
+			if(target && range && speed) //Propel the content if specified.
+				thiscontent.throw_at(target,range,speed,override,fly_speed)
+
+/obj/item/proc/spill_reagents(var/atom/hit_atom) //Spill any reagents contained within the item onto the floor, and the atom it hit when it broke, if applicable.
+	if(!isnull(reagents))
+		if(!isnull(hit_atom) && hit_atom != get_turf(src)) //If it hit something other than the floor, spill it onto that.
+			reagents.reaction(hit_atom, TOUCH)
+		reagents.reaction(get_turf(src), TOUCH) //Then spill it onto the floor.
```

Comment:
Do you need an `else` here? Otherwise the reagents would spill onto a movable first, and then onto the floor. Is this intended? Does it even work? Aren't the reagents consumed by the first `reaction`?

Reply:
The idea is that it first attempts to react anything in the item onto the thing it hit. Then whatever didn't get consumed in the that reaction spills onto the floor. It seems to work from my testing. A beaker full of blood hits a cyborg and the blood then leaves a stain on the floor. A polyacid beaker hits a person and splatters them with acid, burning them.